extensions/azurePublishNew/src/node/provisionService.ts

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { IBotProject } from '@botframework-composer/types';
+
+import { getAppRegistrationProvisionService } from './azureResources/appRegistration';
+import { getWebAppProvisionService } from './azureResources/webApp';
+import { getBotChannelProvisionService } from './azureResources/botChannel';
+import { getAzureFunctionsProvisionService } from './azureResources/azureFunctions';
+import { getCosmosDbProvisionService } from './azureResources/cosmosDb';
+import { getLuisAuthoringProvisionService } from './azureResources/luisAuthoring';
+import { getLuisPredictionProvisionService } from './azureResources/luisPrediction';
+import { getBlogStorageProvisionService } from './azureResources/blobStorage';
+import { getQnAProvisionService } from './azureResources/qna';
+import { getAppServiceProvisionService } from './azureResources/servicePlan';
+
+// bot project => candidate resources => select & configure resources => order & provision
+
+export type ProvisionWorkingSet = Record<string, object>;
```

Comment:
Consider moving the types that are used types.ts files.   /node/types.ts would have types that are used across all node, node/azureResources/types.ts would have types used only across the individual resources (if there are any of those).  The types are compiled away, so the const methods would be put into this module.  This way when resources only depend on types, they don't take an additional module dependency.

Reply:
Another reason for this is you'll avoid with a circular dependency with the resources depending on this for types and this depending on the resources for the get methods.